src/phoebus_boundaries/phoebus_boundaries.cpp

```diff
@@ -136,6 +143,72 @@ void OutflowInnerX1(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
   }
 }
 
+void PolarInnerX2(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
+  std::shared_ptr<MeshBlock> pmb = rc->GetBlockPointer();
+  auto geom = Geometry::GetCoordinateSystem(rc.get());
+
+  auto bounds = coarse ? pmb->c_cellbounds : pmb->cellbounds;
+  PackIndexMap imap;
+  auto q = rc->PackVariables(std::vector<parthenon::MetadataFlag>{Metadata::FillGhost},
+                             imap, coarse);
+  auto nb = IndexRange{0, q.GetDim(4) - 1};
+  auto domain = IndexDomain::inner_x2;
+  const int j0 = bounds.GetBoundsJ(IndexDomain::interior).s;
+
+  auto &fluid = rc->GetParentPointer()->packages.Get("fluid");
+  std::string bc_vars = fluid->Param<std::string>("bc_vars");
+  PARTHENON_REQUIRE(bc_vars == "primitive", "Polar X2 reflecting BCs not supported");
+
+  const auto idx_pvel = imap.GetFlatIdx(fluid_prim::velocity, false);
+  const auto idx_pb = imap.GetFlatIdx(fluid_prim::bfield, false);
+
+  pmb->par_for_bndry(
+      "PolarInnerX2Prim", nb, domain, coarse,
+      KOKKOS_LAMBDA(const int &l, const int &k, const int &j, const int &i) {
+        const int jref = -j + 2 * j0 - 1;
+        if (l == idx_pvel(1)) {
+          q(l, k, j, i) = -q(l, k, jref, i);
+        } else if (l == idx_pb(1)) {
+          q(l, k, j, i) = -q(l, k, jref, i);
```

Comment:
We could use this function (which right now behaves just like the generic BC for prim vars) to implement @Yurlungur's suggestion of only flipping the velocity component. Doing that leads to a stable torus.

Reply:
:+1: